### DISCUSSION TITLE
tls: failed to find any PEM data in key input discovery failed

### DISCUSSION THREAD

**Discussion by rahulbapumore** [etcd-io/etcd#16203] [Q&A]
With installation of etcd as 3 member etcd cluster, inside pod-0 , etcd keeps restarting continuously with error and it kills etcd - 
tls: failed to find any PEM data in key input","message":"discovery failed
`{"caller":"etcdmain/etcd.go:204","error":"tls: failed to find any PEM data in key input","message":"discovery failed","metadata":{"container_name":"dced","namespace":"staging1","pod_name":"dced-0"},"service_id":"dced","severity":"critical","stacktrace":"go.etcd.io/etcd/server/v3/etcdmain.startEtcdOrProxyV2\n\tgo.etcd.io/etcd/server/v3/etcdmain/etcd.go:204\ngo.etcd.io/etcd/server/v3/etcdmain.Main\n\tgo.etcd.io/etcd/server/v3/etcdmain/main.go:40\nmain.main\n\tgo.etcd.io/etcd/server/v3/main.go:32\nruntime.main\n\truntime/proc.go:255","timestamp":"2023-07-10T12:41:36.550+00:00","version":"1.2.0"}`

below is the full log -
[logs.8.38.txt](https://github.com/etcd-io/etcd/files/12003577/logs.8.38.txt)

**Comment by jmhbnz** [comment]
Hey @rahulbapumore - Please confirm what parameters you are using for etcd tls.

For any certificate inputs please verify the certificates are valid using something like `openssl x509 -in <certfile> -noout -text`
